[DOC] Custom resource status (#1678)
strimzi/strimzi-kafka-operator: documentation/book/proc-accessing-resource-status.adoc

// Module included in the following assemblies:
//
// master.adoc

[id='proc-accessing-resource-status-{context}']
= Checking the status of a custom resource

This procedure describes how to find the status of a custom resource.

.Prerequisites

* An {ProductPlatformName} cluster
* A running Cluster Operator

.Procedure

* Specify the custom resource and use `-o jsonpath` option to apply a standard JSONPath expression to select the `status` property:
+
[source,shell,subs="+quotes,attributes"]
----
{cmdcli} get kafka _<kafka_resource_name>_ -o jsonpath='{.status}'
----
+
This expression returns all the status information for the specified custom resource. You can use dot notation, such as `status.listeners`, to fine-tune the status information you wish to see.

.Additional resources
* xref:con-custom-resources-status-{context}[]

> * For more information about using JSONPath, see {KBSJsonPath}.

Reviewer: ```suggestion
* For more information about using JSONPath, see {K8SJsonPath}.
```

// Module included in the following assemblies:
//
// master.adoc

[id='proc-accessing-resource-status-{context}']
= Checking the status of a custom resource

This procedure describes how to find the status of a custom resource.

.Prerequisites

> * An {OpenShiftName} cluster

Reviewer 2: ```suggestion
* An {PlatformProductName} cluster
```

* A running Cluster Operator

.Procedure

> . Specify the custom resource in a standard JSONPath expression to filter on the `status` property:

Reviewer 2: ```suggestion
. Specify the custom resource and use `-o jsonpath` option to apply a standard JSONPath expression to select the `status` property:
```

Reviewer 3: Use a single bullet point for a single-step procedure.

Reviewer 3: Perhaps's Tom's edited text could be split into unordered sub-steps?

Author: > Use a single bullet point for a single-step procedure.

My personal feeling on this is that people will identify a step with a number and it's good to have that consistency for recognition. But I understand if it's the convention to use a bullet for < 2 steps.

Author: > Perhaps's Tom's edited text could be split into unordered sub-steps?

It's a single command, so I guess we leave as a single step.

+
[source,shell,subs="+quotes,attributes"]
----

> {cmdcli} get kafka _kafka_resource_name_ -o jsonpath=”{.status}

Reviewer 2: I don't think you need the quotes at all because `{status}` won't be interpreted by the shell, however since we're likely to include more jsonpath examples in the future and some of them will need to be quoted to prevent shell interpretation we should probably just use quotes everywhere so users get used to seeing and using them.

```suggestion
{cmdcli} get kafka _kafka_resource_name_ -o jsonpath='{.status}'
```

----
+

> This expression returns all the status information. You can use dot notation, such as `status.listeners`, to fine-tune the status information you wish to see.

Reviewer 3: ```suggestion
This expression returns all the status information for the specified custom resource. You can use dot notation, such as `status.listeners`, to fine-tune the status information you wish to see.
```

+

> NOTE: The Kafka bootstrap addresses listed in the status do not signify that those endpoints or the Kafka cluster is in a ready state.

Reviewer 2: This is Kafka-specific, but the procedure title is more generic. Maybe this belongs better in the concept?

.Additional resources

> * For more information about using JSONPath, see {KBSJsonPath}.

Reviewer 3: I suggest adding a link to your new concept here: {ProductName} custom resource status.

Author: We're thinking alike. I just added it as I read this. :wink:
